qtp_job_output_folder/summary.py

```diff
@@ -6,36 +6,64 @@
 # The full license is in the file LICENSE, distributed with this software.
 # -----------------------------------------------------------------------------
 
-from os.path import exists, isdir, join, dirname
-from glob import glob
 from json import dumps
+from os import sep, walk
+from os.path import basename, dirname, exists, isdir, join
 
 
 def _folder_listing(folder):
-    results = []
-    for f in glob(f'{folder}/*'):
-        if isdir(f):
-            results.append(('folder', f))
-            results.extend(_folder_listing(f'{f}/*'))
-        else:
-            results.append(('file', f))
-    return results
+    index, manifest = [], []
+    # only adding main files on top directory
+    # and index.html at any level
+    separator = "|--"
+    for dpath, _, files in walk(folder):
+        # assuring same order, mainly for testing
+        files.sort()
+
+        # if we are at the top, we should add
+        # all files
+        if dpath == folder:
+            for f in files:
+                index.append(("file", f"{dpath}/{f}"))
+        # if we are not at the top, we should only add
+        # the index.html files
+        elif "index.html" in files:
+            index.append(("file", f"{dpath}/index.html"))
+
+        depth = dpath.replace(folder, "").count(sep)
```

Comment:
is folder relative or absolute? What if the path contains the folder infix twice like `job-output/4/job-output/4/subdir`?